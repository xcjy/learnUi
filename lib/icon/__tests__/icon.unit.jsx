import * as renderer from "react-test-renderer";
import Icon from "../icon.tsx";
import {mount} from 'enzyme'
import React from "react";
describe("icon tests", () => {
    it('icon test',()=>{
        const json= renderer.create(<Icon name="alipay"></Icon>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('onClick',()=>{
        const fn= jest.fn()
        const component= mount(<Icon name='alipay' onClick={fn}/>)
        component.find('svg').simulate('click')
        expect(fn).toBeCalled()
})})
