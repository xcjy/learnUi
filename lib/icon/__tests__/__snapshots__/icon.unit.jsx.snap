// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon tests icon test 1`] = `
<svg
  className="learnUi-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
